python/scripts/overwrite-dependency.py

```diff
@@ -0,0 +1,138 @@
+    """
+    FOSSA generates the s3 signed url
+    """
+    try:
+        url = 'https://app.fossa.com/api/components/signed_url?packageSpec={packageSpec}&revision={revisionVersion}'.format(packageSpec=packageSpec, revisionVersion=revisionVersion)
+        response = requests.get(url, headers={"Authorization": "Bearer %s" %os.environ['FOSSA_API_KEY']})
+        signedUrl = response.json().get('signedUrl')
+        logger.info("Got a presigned response: %s", signedUrl)
+
+    except ClientError:
+        logger.exception("Couldn't get a presigned URL for client method")
+        raise
+    return signedUrl
+
+def build_dependency(title, version, description, dependencyUrl):
+    """
+    FOSSA builds the uploaded dependency
+    """
+    try:
+        fossa_api_token = os.environ['FOSSA_API_KEY']
+        url_endpoint = 'https://app.fossa.com/api/components/build'
+
+        headersAuth = {
+            'Authorization': 'Bearer ' + fossa_api_token,
+            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
+        }
+
+        params = (
+            ('description', description),
+            ('projectURL', dependencyUrl),
+            ('title', title),
+            ('dependency', 'true'),
+            ('rawLicenseScan', 'true')
+        )
+
+        data = 'archives%5B0%5D%5BpackageSpec%5D={packageSpec}&archives%5B0%5D%5Brevision%5D={version}'.format(packageSpec=title, version=version)
+
+        response = requests.post(url_endpoint, headers=headersAuth, params=params, data=data, verify=True)
+        logger.info("Built dependency.... %s", response.status_code)
+
+    except HTTPException:
+        logger.exception("Couldn't do it, fail")
+        raise
+
+def replace_dependency_in_project(projectLocator, name, version, homepage, description, overwrittenDepLocator):
+        """
+        FOSSA replaces the dependency
+        """
+        try:
+            fossa_api_token = os.environ['FOSSA_API_KEY']
+            url_endpoint = 'https://app.fossa.com/api/revisions/{projectLocator}/dependencies'.format(projectLocator=quote(projectLocator,safe=''))
+
+            headersAuth = {
+                'Authorization': 'Bearer ' + fossa_api_token,
+                'Content-Type': 'application/json'
+            }
+
+            data = {
+                "dependencyData": {
+                    "kind":"replace",
+                    "newDependency": {
+                        "kind":"archive",
+                        "name": name,
+                        "version": version,
+                        "homepage": homepage,
+                        "description": description
+                    },
+                    "overwrittenLocator": overwrittenDepLocator
+                }
+            }
+
+            response = requests.post(url_endpoint, headers=headersAuth, json=data, verify=True)
+            logger.info("Overwrote dependency in project: %s", response.status_code)
```

Comment:
There's no handle here if the overwrite is attempted more than once. If the previous steps were successful and overwriting happens more than once, then it appears the dependency moved from unknown to incomplete and the status code is 400.